### PR TITLE
[WFLY-11566] Fix EJB validation - provide information about generics to validator

### PR DESCRIPTION
**Pull request by tadamski**
https://issues.redhat.com/browse/WFLY-11566
https://issues.redhat.com/browse/JBEAP-16121

Please make sure your PR meets the following requirements:
- [ ] Pull Request title is properly formatted: `[WFLY-XYZ] Subject` or `WFLY-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue(s)
- [ ] Pull Request contains description of the issue(s)
- [ ] Pull Request does not include fixes for issues other than the main ticket
- [ ] Attached commits represent units of work and are properly formatted

For bigger changes, major and minor component upgrades make sure your PR also meets following requirements:
- [ ] Pull Request requires a change to the documentation
- [ ] Documentation have been updated accordingly
- [ ] Tests were added to cover changes

For new features ensure as well:
- [ ] Analysis was done
- [ ] Test Plan has been done
- [ ] Tests were verified in advance

If you are not an active contributor of the WildFly project you can request sponsorship by one of the members to help guide you through the process.

### CODE REVIEW

**Comment by bstansberry** [comment]
HI @tadamski 

These new dependencies need to be guarded by a check for org.wildfly.bean-validation.  The module dependencies should be optional. The runtime code should not load any classes from the optional modules unless the presence of the capability has been confirmed.

File: ee/src/main/java/org/jboss/as/ee/component/DefaultComponentViewConfigurator.java [hunk 1]
```diff
@@ -78,6 +78,14 @@ public void configure(final DeploymentPhaseContext context, final ComponentDescr
                 }
             }
 
+            Class<?> markupClass;
+            try {
+                markupClass = module.getClassLoader().loadClass(view.getMarkupClassName());
```
Comment (bstansberry) [inline ee/src/main/java/org/jboss/as/ee/component/DefaultComponentViewConfigurator.java:83]: This should handle the possibility that view.getMarkupClassName() will return null.

File: ejb3/src/main/java/org/jboss/as/ejb3/validator/EjbProxyBeanMetaDataClassNormalizer.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    }
+
+    @Override
+    public void destroy(BeanMetaDataClassNormalizer beanMetaDataClassNormalizer, CreationalContext<BeanMetaDataClassNormalizer> creationalContext) {
+
+    }
+
+    @Override
+    public Set<Type> getTypes() {
+        return Collections.singleton( BeanMetaDataClassNormalizer.class );
```
Comment (bstansberry) [inline ejb3/src/main/java/org/jboss/as/ejb3/validator/EjbProxyBeanMetaDataClassNormalizer.java:80]: If this is in the request handling path please use a constant.

File: ejb3/src/main/java/org/jboss/as/ejb3/validator/EjbProxyBeanMetaDataClassNormalizer.java [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    }
+
+    @Override
+    public Set<Annotation> getQualifiers() {
+        return Collections.singleton( NamedLiteral.of( BeanNames.BEAN_META_DATA_CLASS_NORMALIZER ) );
```
Comment (bstansberry) [inline ejb3/src/main/java/org/jboss/as/ejb3/validator/EjbProxyBeanMetaDataClassNormalizer.java:85]: If this is in the request handling path please use a constant.

**Comment by tadamski** [comment]
hi @bstansberry 
I have fixed the issues mentioned

**Comment by bstansberry** [comment]
Hi @tadamski Thanks!

My only other request is that you add some class level javadoc for the key new classes. It's not obvious what these classes are doing so some years from now that will save a maintainer time.

If you have someone in mind who better understands the issue, please @ ping them for a review. Otherwise just @ ping me and we'll move it along.

**Comment by tadamski** [comment]
I have added the javadoc
@chengfang can you give a look on this?

**Review by bstansberry — APPROVED** [review]
Thanks @tadamski

**Review by chengfang — APPROVED** [review]

**Pull request merged**